src/odemis/driver/test/pigcs_test.py

```diff
@@ -601,6 +602,47 @@ def setUp(self):
         self.kwargs = KWARGS_CL
         self.kwargs_two = KWARGS_TWO_CL
 
+    #    @skip("faster")
+    def test_reference(self):
+        """
+        Test referencing for 2 axes.
+        """
+        stage = CLASS(**self.kwargs_two)
+
+        # Test proper referencing
+        self.assertTrue(stage.referenced.value['x'])
+        self.assertTrue(stage.referenced.value['y'])
+        pos = stage.position.value
+        stage.moveAbs({'x': 0.015}).result()
+        stage.moveAbs({'y': 0.005}).result()
+        self.assertAlmostEqual(stage.position.value['x'], 0.015, places=3)  # 0.001 times out
+        self.assertAlmostEqual(stage.position.value['y'], 0.005, places=3)
+        self.assertNotAlmostEqual(stage.position.value['x'], pos["x"], places=3)
+
+        stage.reference({'x'}).result()
+        self.assertTrue(stage.referenced.value['x'])
+        # The reference position should be 12.5 mm for simulator
+        self.assertAlmostEqual(stage.position.value["x"], 0.0125, places=3)
```

Comment:
The problem with such check is that it only works with the simulator. We could try to read the parameter, and convert it using the *right* conversion factor. Alternatively, you could just do it twice (with a move in between) and check it's the same value.